### ISSUE TITLE
[naming-convention] Support destructured modifier for parameters

### ISSUE THREAD

**Issue by abacabadabacaba** [typescript-eslint/typescript-eslint#2828]
Currently, `naming-convention` rule supports `destructured` modifier only for variables. I propose to also support it for parameters.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": ["error",
      {
        "selector": "parameter",
        "modifiers": ["destructured"],
        "format": ["camelCase"]
      }
    ]
  }
}
```

```TS
function foo({ aNameToBeChecked }: any) {
  // ...
}
```

**Expected Result**

The identifier `aNameToBeChecked` matches the rule.

**Actual Result**

The `destructured` modifier isn't supported with `selector: "parameter"`, so it doesn't work.

**Versions**

This issue is currently present in the master branch.

**Comment by bradzacher** [comment]
seems reasonable

**Comment by abacabadabacaba** [comment]
@bradzacher I think that this modifier should be supported for `selector: "variableLike"` as well, as it includes both `variable` and `parameter`.

**Comment by bradzacher** [comment]
You can use the array form of selector to specify multiple selectors.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#selector-options

**Comment by abacabadabacaba** [comment]
> You can use the array form of selector to specify multiple selectors.

Yes, but the whole point of group selectors is to be a concise replacement for all the selectors they are equivalent to, so they should support the same modifiers as the individual selectors.

Currently, it is possible to use `modifiers: ["destructured"]` with `selector: "default"` and `selector: "variable"` but not `selector: "variableLike"`, and it doesn't seem right.
